python/aws_lambda_powertools/logging/logger.py

```diff
@@ -55,12 +63,25 @@ def logger_setup(service: str = "service_undefined", level: str = "INFO", **kwar
 
     """
     service = os.getenv("POWERTOOLS_SERVICE_NAME") or service
+    sampling_rate = os.getenv("POWERTOOLS_LOGGER_SAMPLE_RATE") or sampling_rate
     log_level = os.getenv("LOG_LEVEL") or level
     logger = logging.getLogger(name=service)
+
+    try:
+        if sampling_rate and random.random() <= float(sampling_rate):
+            log_level = logging.DEBUG
+    except ValueError:
+        raise ValueError(
+            f"Expected a float value ranging 0 to 1, but received {sampling_rate} instead. Please review "
+            f"POWERTOOLS_LOGGER_SAMPLE_RATE environment variable."
```

Comment:
You can merge this line with the above instead of having two lines ;)

f"Expected a float value ranging 0 to 1, but received {sampling_rate} instead. Please review POWERTOOLS_LOGGER_SAMPLE_RATE environment variable."